hashring/hashring_test.go

```diff
@@ -352,3 +352,19 @@ func BenchmarkHashRingLookupN(b *testing.B) {
 		}
 	}
 }
+
+func TestConsistentOnCollision(t *testing.T) {
+	// make 11 replica points so that 127.0.0.1:3010 collides
+	r1 := New(farm.Fingerprint32, 11)
+	r2 := New(farm.Fingerprint32, 11)
+
+	r1.AddServer("127.0.0.1:30")
+	r1.AddServer("127.0.0.1:301")
+	r2.AddServer("127.0.0.1:301")
+	r2.AddServer("127.0.0.1:30")
+
+	m1, _ := r1.Lookup("127.0.0.1:3010")
+	m2, _ := r2.Lookup("127.0.0.1:3010")
+
+	assert.Equal(t, m1, m2, "hashring isn't consistent due to hash collisions")
```

Comment:
I would prefer to actually compare the full hash rings for completeness if possible. (also my previous comment on the assert still holds up).